### PR TITLE
Ide phpstorm debugging

### PR DESCRIPTION
**Pull request by marcorensch**
### **User description**
Added an introduction on how to setup Xdebug with MAMP & PhpStorm.

___

### **PR Type**
documentation

___

### **Description**
- Added a detailed guide on setting up Xdebug with MAMP and PhpStorm for Joomla! extensions debugging.
- Updated documentation to correct the casing of PhpStorm.
- Improved links to PhpStorm documentation for better navigation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.md</strong><dd><code>Update PhpStorm documentation link and casing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/get-started/ide/index.md

<li>Updated link to PhpStorm documentation.<br> <li> Corrected the casing of PhpStorm.<br>

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/344/files#diff-47280ad5f1faa16e77ed146a4a9bb31c083d46697ba13c44f223573872a29be5">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>debugging.md</strong><dd><code>Add guide for PhpStorm debugging with MAMP and Xdebug</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/get-started/ide/phpstorm/debugging.md

<li>Added a comprehensive guide for setting up Xdebug with MAMP and <br>PhpStorm.<br> <li> Included instructions for enabling Xdebug in MAMP.<br> <li> Provided steps for configuring PhpStorm debugger.<br> <li> Added images and notes for clarity.<br>

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/344/files#diff-50ad882112f9bf17e7cd068560a09a8fffc201afc4f063bc9fb41b5d2b15dfae">+142/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.md</strong><dd><code>Correct PhpStorm casing in title</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/get-started/ide/phpstorm/index.md

- Corrected the casing of PhpStorm in the title.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/344/files#diff-937baf1a2909753ee14fb6994d889ec917fd4b7a78701ec624e32b0cc94b22ee">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: docs/get-started/ide/phpstorm/debugging.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+Debugging
+========
```
Comment (HLeithner) [inline docs/get-started/ide/phpstorm/debugging.md:2]:
```suggestion
=========
```

File: docs/get-started/ide/phpstorm/debugging.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+![MAMP PHP INI Settings](./assets/php_ini_setup.png)
+
+Hint: Use `Ctrl` + `F` and search for "xdebug"
+
+:::note
+
+After you have changed some settings make sure to save & restart your MAMP Server.
+
+:::
```
Comment (HLeithner) [inline docs/get-started/ide/phpstorm/debugging.md:33]:
```suggestion
:::note
  After you have changed some settings make sure to save & restart your MAMP Server.
:::
```

File: docs/get-started/ide/phpstorm/debugging.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+So beside your extension development project in PhpStorm open or create a new Project for Debugging inside your local
+Joomla! Installation. This Project should point to the Joomla! Main directory of your local Joomla! Installation. So in
+PhpStorm click on `File` > `New Project form Existing Files`and select your Joomla! Installation. PhpStorm may ask you
+if you would like to Trust this Folder, confirm that request.
+
+### Configure PhpStorm Debugger
+
+In this new Project select `Run` from the Main Menu and Select `Edit Configurations`.
+
+![Edit Configurations Menu](./assets/run_edit_configurations.png)
```
Comment (HLeithner) [inline docs/get-started/ide/phpstorm/debugging.md:64]:
```suggestion
![Edit Configurations Menu](./assets/run_edit_configurations.png)
```

Please move all images from "assets" to "_assets" folder

File: docs/get-started/ide/phpstorm/debugging.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+Tick the option `Filter debug connection by IDE Key` and click on the Button with the three dots beside the Server
+Selection.
+
+![Select remote debug](./assets/start_configuration.png)
+
+This will open a new PopUp window to set up the server settings. Enter here your MAMP's Server Information based on your
+MAMP Settings.
+The name can be "MAMP Server" or whatever you like. By Default MAMP Server's are running on Port 8890 use the Port of
+your Server here. In this case here the Joomla! Installation is available on the URL `https://joomla5.local:8890` so the
+Host Name is `joomla5.local`. Set now the Hostname and Port based on your configuration.
```
Comment (HLeithner) [inline docs/get-started/ide/phpstorm/debugging.md:84]: .local is a reserved tld top level domain which is used for mdns.

please use joomla5.example.com for documentation or .internal for real internal dns usage.

**Review by HLeithner — CHANGES_REQUESTED** [review]
Do you have links to existing documentation? so that even if this documentation is outdated the read might have a resource with an update documentation, maybe provided by phpstorm or xdebug?

Please add such links to documentation.

File: docs/get-started/ide/phpstorm/debugging.md (hunk 3 above)
Comment (marcorensch) [reply docs/get-started/ide/phpstorm/debugging.md]: done

File: docs/get-started/ide/phpstorm/debugging.md (hunk 4 above)
Comment (marcorensch) [reply docs/get-started/ide/phpstorm/debugging.md]: right... used now "joomla5.localhost" because the guide implies local development with MAMP

**Comment by marcorensch** [comment]
Added ext ressources links at the end of the article and implemented the requested changes.

File: docs/building-extensions/templates/template-details-file.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ The sections are defined as:
   <namespace><!-- Description of the template go here, usually CompanyNamespace\Templates\Templatename --></namespace>
 	<inheritable>1</inheritable>
 	<files><!-- Files/Folders for the template folder entries go here --></files>
-	<media destination="templates/site/cassiopeia" folder="media"><!-- Folders for the static assets entries go here --></media>
+	<media destination="templates/site/cassiopeia" folder="media"><!-- Folders for the static _assets entries go here --></media>
```
Comment (HLeithner) [inline docs/building-extensions/templates/template-details-file.md:20]: Sorry it seems I miss communicated or you miss understood me, only in the manual we need to use the `_assets` folder not in joomla it self.

Comment (marcorensch) [reply docs/building-extensions/templates/template-details-file.md:20]: nope that was clearly a mistake...

**Review by HLeithner — APPROVED** [review]

**Comment by HLeithner** [comment]
thanks

**Pull request merged**